importer/trickledag.go

```diff
@@ -0,0 +1,91 @@
+package importer
+
+import (
+	"io"
+
+	"github.com/jbenet/go-ipfs/importer/chunk"
+	dag "github.com/jbenet/go-ipfs/merkledag"
+	"github.com/jbenet/go-ipfs/pin"
+)
+
+// layerRepeat specifies how many times to append a child tree of a
+// given depth. Higher values increase the width of a given node, which
+// improves seek speeds.
+const layerRepeat = 4
+
+func BuildTrickleDagFromReader(r io.Reader, ds dag.DAGService, mp pin.ManualPinner, spl chunk.BlockSplitter) (*dag.Node, error) {
+	// Start the splitter
+	blkch := spl.Split(r)
+
+	// Create our builder helper
+	db := &dagBuilderHelper{
+		dserv:    ds,
+		mp:       mp,
+		in:       blkch,
+		maxlinks: DefaultLinksPerBlock,
+		indrSize: defaultIndirectBlockDataSize(),
+	}
```

Comment:
let's not share datstructures here. changing the regular dag reader can break trickle, etc. why dont we separate into packages:

```
/importer      <--- maybe exports a version of dagBuilderHelper that subpkgs can use
/importer/balanced   <--- the normal one
/importer/trickle
```